<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>627</width>
    <height>491</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <property name="sizeConstraint">
    <enum>QLayout::SetMinimumSize</enum>
   </property>
   <item row="0" column="0">
    <layout class="QGridLayout" name="mainLayout">
     <property name="sizeConstraint">
      <enum>QLayout::SetMaximumSize</enum>
     </property>
     <item row="0" column="0">
      <layout class="QVBoxLayout" name="verticalLayout">
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Insert URL to download:</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLineEdit" name="urlLineEdit">
           <property name="toolTip">
            <string>Insert valid download link</string>
           </property>
           <property name="placeholderText">
            <string>https://...</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="chooseLocationButton">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="toolTip">
            <string>Choose saving location</string>
           </property>
           <property name="text">
            <string>Save as</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="startIndividualDownloadButton">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="toolTip">
            <string>Start download from URL</string>
           </property>
           <property name="text">
            <string>Start download</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QPushButton" name="startSelectedDownloadButton">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="toolTip">
            <string>Start selected downloads</string>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset>
             <normaloff>../icons/play.svg</normaloff>../icons/play.svg</iconset>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pauseSelectedDownloadButton">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="toolTip">
            <string>Pause selected downloads</string>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset>
             <normaloff>../icons/pause.svg</normaloff>../icons/pause.svg</iconset>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="cancelSelectedDownloadButton">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="toolTip">
            <string>Stop selected downloads</string>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset>
             <normaloff>../icons/stop.svg</normaloff>../icons/stop.svg</iconset>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>pauseSelectedDownloadButton</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>pause_download()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>418</y>
    </hint>
    <hint type="destinationlabel">
     <x>181</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>startSelectedDownloadButton</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>start_resume_download()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>75</x>
     <y>418</y>
    </hint>
    <hint type="destinationlabel">
     <x>351</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chooseLocationButton</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>choose_location_save()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>476</x>
     <y>259</y>
    </hint>
    <hint type="destinationlabel">
     <x>536</x>
     <y>5</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>startIndividualDownloadButton</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>start_individual_download()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>612</x>
     <y>259</y>
    </hint>
    <hint type="destinationlabel">
     <x>515</x>
     <y>9</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cancelSelectedDownloadButton</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>cancel_download()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>173</x>
     <y>415</y>
    </hint>
    <hint type="destinationlabel">
     <x>160</x>
     <y>4</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>urlLineEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>Form</receiver>
   <slot>parse_url()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>234</x>
     <y>252</y>
    </hint>
    <hint type="destinationlabel">
     <x>316</x>
     <y>4</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>start_resume_download()</slot>
  <slot>pause_download()</slot>
  <slot>choose_location_save()</slot>
  <slot>start_individual_download()</slot>
  <slot>cancel_download()</slot>
  <slot>parse_url()</slot>
 </slots>
</ui>
